packages/components/src/presentation-components/table/UsePresentationTable.ts

```diff
@@ -84,10 +96,59 @@ export function usePresentationTable<TColumn, TRow>(props: UsePresentationTableP
  */
 export function usePresentationTableWithUnifiedSelection<TColumn, TRow>(
```

Comment:
Current implementation does not handle selection of lazy loaded rows. For example:
- Table loads 10 rows.
- Other component updates unified selection and puts instance key of instance representated by the row 11 which is not yet loaded.
- Table is scrolled and 11 row is loaded.
- `selectedRows` returns by this hook should include 11th row.

Reply:
This was fixed by additionally updating `selectedRows` when the `rows` variable changes. So now every time new rows get loaded, these new rows get checked for having matching keys that exist in the `unifiedSelection`.